File: cpp/Array/L_subArrSumDivByK.cpp
#include<bits/stdc++.h>
using namespace std;



int subarraysDivByK(vector<int>& nums, int K) {
    vector<int> count(K);
    count[0] = 1;
    int prefix = 0, res = 0;
    for (int a : nums) {
        prefix = (prefix + a % K + K) % K;
        res += count[prefix]++;
    }
    return res;
}

File: JavaScript/Array/L_subArrSumDivByK.js
var subarraysDivByK = function (nums, k) {
    let res = 0;
    let prefix = 0;
    let arr = new Array(k).fill(0);
    arr[0] = 1;

    for (let num of nums) {
        prefix = (prefix + (num % k) + k) % k;
        res += arr[prefix];
        arr[prefix]++;
    }
    return res;
};
